### PR TITLE
F/callout layer

### PR DESCRIPTION
**Pull request by Brynjulf**
Initial work for callout labeling #11

### CODE REVIEW

File: src/layers/CanvasLayer.ts [hunk 1]
```diff
@@ -18,6 +18,7 @@ export abstract class CanvasLayer extends Layer {
       this.canvas = canvas;
       event.elm.appendChild(canvas)
     }
+    this.canvas.setAttribute('id', this.id.toString());
```
Comment (stalenestas) [inline src/layers/CanvasLayer.ts]: I think eslint would complain if we don't use `${this.id}`

File: src/layers/CanvasLayer.ts [hunk 2]
```diff
@@ -34,9 +35,13 @@ export abstract class CanvasLayer extends Layer {
...
     } = this;
+    const {
+      xScale,
+      yScale,
+    } = event;
 
     ctx.canvas.setAttribute('style', `position:absolute;z-index:${this.order};opacity:${this.opacity}`);
-    ctx.canvas.setAttribute('width', `${elm.getAttribute('width')}px`)
-    ctx.canvas.setAttribute('height', `${elm.getAttribute('height')}px`);
+    ctx.canvas.setAttribute('width', `${xScale.range()[1]}px`)
```
Comment (stalenestas) [inline src/layers/CanvasLayer.ts]: I think 
const [, width] = xScale.range() 
is used elsewhere

File: src/layers/CalloutCanvasLayer.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+  private renderPoint(x : number, y : number) {
+    this.ctx.moveTo(x, y);
+    this.ctx.arc(x, y, 4, 0, Math.PI * 2);
+    this.ctx.fill();
+  }
+
+  private renderLines(x : number, y : number, offset : number, title : string) {
+    this.ctx.moveTo(x, y);
+    this.ctx.lineTo(x - offset, y + offset + 2);
```
Comment (stalenestas) [inline src/layers/CalloutCanvasLayer.ts]: group x-offset to a const above

File: src/layers/CanvasLayer.ts (hunk 1 above)
Comment (Brynjulf) [reply src/layers/CanvasLayer.ts]: i believe that rule applies to strings that are concatenated, but i'll change it nonetheless

**Pull request merged**